kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaLogActionUtils.java

```diff
@@ -266,7 +269,11 @@ public static CloseableIterator<FileStatus> listDeltaLogFilesAsIter(
                 }
               }
 
-              hasReturnedAnElement.set(true);
+              // Only log and checkpoint could use to construct table state.
+              if (FileNames.isCommitFile(getName(fs.getPath()))
```

Comment:
I just realized this. `isCommitFile` takes in a String path. so does `isCheckpointFile`. Want to update all of these occurences?

It can just be : `FileNames.isCommitFile(fs.getPath())`